File: app/src/main/java/com/mapelli/simone/githubclient/viewmodel/UserSearchViewModelFactory.java
package com.mapelli.simone.githubclient.viewmodel;


import com.mapelli.simone.githubclient.UI.UsersSearchActivity;

import androidx.annotation.NonNull;
import androidx.lifecycle.ViewModel;
import androidx.lifecycle.ViewModelProvider;

public class UserSearchViewModelFactory extends ViewModelProvider.NewInstanceFactory {
    private final static String TAG = UsersSearchActivity.class.getSimpleName();


    @NonNull
    @Override
    public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {
        return (T) new UserSearchViewModel();
    }
}

File: app/src/main/java/com/mapelli/simone/githubclient/viewmodel/UserSearchViewModelFactory.kt
package com.mapelli.simone.githubclient.viewmodel


import com.mapelli.simone.githubclient.UI.UsersSearchActivity
import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider

class UserSearchViewModelFactory : ViewModelProvider.NewInstanceFactory() {


    override fun <T : ViewModel> create(modelClass: Class<T>): T {
        return UserSearchViewModel() as T
    }

    companion object {
        private val TAG = UsersSearchActivity::class.java!!.getSimpleName()
    }
}
